osidb/tests/test_flaw.py

```diff
@@ -748,3 +755,39 @@ def test_validate_reported_date_non_empty(self):
         # whenever we save the flaw which the factory does automatically the validations are run
         # and if there is an exception the test will fail so creating the flaw is enough to test it
         FlawFactory()
+
+    @pytest.mark.parametrize(
+        "embargoed,unembargo_date,should_raise",
+        [
+            (False, None, True),
+            (False, tzdatetime(2021, 11, 22), False),
+            (True, None, False),
+            (True, tzdatetime(2021, 11, 22), False),
+        ],
+    )
+    @freeze_time(tzdatetime(2021, 11, 23))
+    def test_validate_public_unembargo_date(
+        self, embargoed, unembargo_date, should_raise
+    ):
+        if should_raise:
+            with pytest.raises(ValidationError) as e:
+                FlawFactory(unembargo_dt=unembargo_date, embargoed=embargoed)
+            assert "Public flaw has an empty unembargo_dt" in str(e)
+        else:
+            assert FlawFactory(unembargo_dt=unembargo_date, embargoed=embargoed)
+
+    @freeze_time(tzdatetime(2021, 11, 23))
+    def test_validate_future_unembargo_date(self):
+        """test unembargo_dt validation - no future date"""
+        future_dt = tzdatetime(2021, 11, 27)
+
+        with pytest.raises(ValidationError) as e:
+            FlawFactory(unembargo_dt=future_dt)
```

Comment:
you don't explicitly set `embargoed` here and by default it's random so this test will fail 50% of the time

Reply:
sorry, there were some other tests with the same problem, I have updated it.